### PR TITLE
Datadog Metric Slicing

### PR DESCRIPTION
**Pull request by Jtang-1**
## Product Description

For the metric `formplayer.metrics.timings`
-  Datadog tag `module_initiated_by` is renamed to `request_initiated_by`.
- `app_name` tag is added
- `module_name` tag is included when `navigate_menu` enters a menu

TODO: Add screenshot

Screen Shot 2024-03-18 at 4 46 21 PM

## Technical Summary

[Jira Ticket ](https://dimagi.atlassian.net/browse/USH-4182)
[Related HQ Changes](https://github.com/dimagi/commcare-hq/pull/34247)

This PR consists of two changes:
1. The `module_initiated_by` tag was introduced previously in this [PR](https://github.com/dimagi/formplayer/pull/1533). This tag will be renamed to a more general `request_initiated_by`. This name better reflects that the tag reflects the user action that resulted in the request. 
2. The `request_initiated_by` value is moved from `QueryData` to `SessionNavigationBean` as the values are no longer specific to  just `Query`

### Safety story

tested on staging
The blast zone would include endpoints where HQ includes a `requestInitiatedByTag` value. However, the logic added only involves adding a datadog tag which is already extensively used.

### Automated test coverage

tests already exists for `navigate_menu` and `navigate_menu_start` endpoints.

### QA Plan

[QA Ticket](https://dimagi.atlassian.net/browse/QA-6188?focusedCommentId=316724&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316724)

### Special deploy instructions

- [X] This PR can be deployed after merge with no further considerations.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations.

### Review

- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### CODE REVIEW

**Review by MartinRiese — APPROVED** [review]
Looks pretty straight forward

File: src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java [hunk 1]
```diff
@@ -53,6 +56,15 @@ public Object logRequest(ProceedingJoinPoint joinPoint) throws Throwable {
             domain = bean.getDomain();
         }
 
+        if (args != null && args.length > 0 && args[0] instanceof SessionNavigationBean) {
+            SessionNavigationBean bean = (SessionNavigationBean) args[0];
+            String requestInitiatedByTag = bean.getRequestInitiatedByTag();
+            if (requestInitiatedByTag != null) {
+                System.out.println("requestedInitiateydby is " + requestInitiatedByTag);
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java:63]: do we need this for prod ?

File: src/main/java/org/commcare/formplayer/application/MenuController.java [hunk 2]
```diff
@@ -61,6 +62,9 @@ public class MenuController extends AbstractBaseController {
     @Autowired
     private ResponseMetaDataTracker responseMetaDataTracker;
 
+    @Autowired
+    private FormplayerDatadog datadog;
```
Comment (shubham1g5) [inline src/main/java/org/commcare/formplayer/application/MenuController.java:66]: don't see it getting used in this class..

**Review by shubham1g5 — APPROVED** [review]

File: src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java (hunk 1 above)
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/aspects/MetricsAspect.java]: no, deleted here [5feea3c](https://github.com/dimagi/formplayer/pull/1551/commits/5feea3c80f5ed361b9add353d38d52757afd994a)

File: src/main/java/org/commcare/formplayer/application/MenuController.java (hunk 2 above)
Comment (Jtang-1) [reply src/main/java/org/commcare/formplayer/application/MenuController.java]: ah i moved the code that used that to `MetricAspect`. Deleted here [5feea3c](https://github.com/dimagi/formplayer/pull/1551/commits/5feea3c80f5ed361b9add353d38d52757afd994a)

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**